pkg/shp/follower/follower.go

```diff
@@ -0,0 +1,219 @@
+package follower
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/shipwright-io/cli/pkg/shp/params"
+	"github.com/shipwright-io/cli/pkg/shp/reactor"
+	"github.com/shipwright-io/cli/pkg/shp/tail"
+	"github.com/shipwright-io/cli/pkg/shp/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+)
+
+// PodLogsFollower uses PodWatcher to react upon pod state changes with the final objective of
+// streaming all logs from all the containers in the pod.
+type PodLogsFollower struct {
+	ctx context.Context
+
+	pw        *reactor.PodWatcher          // podwatcher instance
+	clientset kubernetes.Interface         // kubernetes client
+	ioStreams *genericclioptions.IOStreams // log output interface
+
+	podIsRunning   bool            // pod is set as running
+	logTail        *tail.Tail      // helper to tail container logs
+	logTailStarted map[string]bool // containers with logTail started
+
+	onlyOnce bool // retrieve the pod logs only once
```

Comment:
This irritated me a little bit. If I understand it correctly, then this PodLogsFollower is capable to retrieve logs with and without following which is controller by this flag. If that's the case, what about

* renaming `PodLogsFollower` to `PodLogsRetriever`
* renaming `onlyOnce` to `follow`

Alternatively, we could also just change `onlyOnce` to `noFollow` as a simple solution (but then the struct name would be confusing).